src/main/java/ch/heigvd/pro/b04/questions/QuestionController.java

```diff
@@ -171,19 +175,99 @@ public ServerQuestion insert(
+   * @throws ResourceNotFoundException if one of the parameters is broken
+   */
+  @PutMapping(value = "/mod/{idModerator}/poll/{idPoll}/question/{idQuestion}")
+  @Transactional
+  public ServerQuestion updateQuestion(
+      @RequestParam(name = "token") String token,
+      @PathVariable(name = "idModerator") int idModerator,
+      @PathVariable(name = "idPoll") int idPoll,
+      @PathVariable(name = "idQuestion") int maggieQ,
+      @RequestBody ClientQuestion question
+  ) throws WrongCredentialsException, ResourceNotFoundException {
+    if (hasAccessAsModerator(idModerator, token).isEmpty()) {
+      throw new WrongCredentialsException();
+    }
+
+    ServerQuestion upQ = fetchQuestionByPollAndModo(idModerator, token, idPoll, maggieQ);
+
+    upQ.setTitle(question.getTitle());
+    upQ.setDetails(question.getDetails());
+    upQ.setVisibility(question.getVisibility());
+    upQ.setAnswersMin(question.answersMin);
+    upQ.setAnswersMax(question.answersMax);
+
+    return upQ;
+  }
+
+  /**
+   * Update a {@link ServerQuestion} in a {@link ServerPoll}.
+   *
+   * @param token token of the moderator
+   * @param idModerator id of the moderator
+   * @param idPoll id of the poll owning the question
+   * @param maggieQ id of the question to delete
+   * @return confirmation message
+   * @throws ResourceNotFoundException if one of the parameters is broken
+   */
+  @DeleteMapping(value = "/mod/{idModerator}/poll/{idPoll}/question/{idQuestion}")
+  @Transactional
+  public ServerMessage deleteQuestion(
+      @RequestParam(name = "token") String token,
+      @PathVariable(name = "idModerator") int idModerator,
+      @PathVariable(name = "idPoll") int idPoll,
+      @PathVariable(name = "idQuestion") int maggieQ) throws ResourceNotFoundException {
+    repository.delete(fetchQuestionByPollAndModo(idModerator, token, idPoll, maggieQ));
+    return ServerMessage.builder().message("Question deleted").build();
+  }
+
+  /**
+   * fetch a {@link ServerQuestion} and verify its access by doing it.
+   *
+   * @param idModo id of the moderator
+   * @param token token of the moderator
+   * @param idPoll id of the poll owning the question
+   * @param idQuestion id of the question to retrieve
+   * @return {@link ServerQuestion} found
+   * @throws ResourceNotFoundException if question cannot be found
+   */
+  private ServerQuestion fetchQuestionByPollAndModo(int idModo, String token,
+      int idPoll, int idQuestion) throws ResourceNotFoundException {
+    ServerPoll poll = pollRepository.findById(ServerPollIdentifier.builder()
+        .idxModerator(hasAccessAsModerator(idModo, token).get())
+        .idPoll(idPoll)
+        .build())
+        .orElseThrow(ResourceNotFoundException::new);
+
+    ServerQuestion upQ = repository.findById(ServerQuestionIdentifier.builder()
+        .idServerQuestion(idQuestion).idxPoll(poll).build())
+        .orElseThrow(ResourceNotFoundException::new);
+
+    return upQ;
+  }
```

Comment:
This should probably be in `ServerQuestion` and set as `static`

Reply:
I think it makes more sense if it's in the same file, grouped with the first ones